### PR TITLE
Remove dead code in rest and http packages

### PR DESCRIPTION
**Pull request by original-brownbear**
Removing some more unused and needlessly verbose things here,
this code is complicated enough as it is already.

### CODE REVIEW

**Review by idegtiarenko — APPROVED** [review]

File: server/src/main/java/org/elasticsearch/http/HttpHandlingSettings.java [hunk 1]
```diff
@@ -71,54 +39,7 @@ public static HttpHandlingSettings fromSettings(Settings settings) {
             SETTING_HTTP_RESET_COOKIES.get(settings),
             SETTING_HTTP_COMPRESSION.get(settings),
             SETTING_HTTP_COMPRESSION_LEVEL.get(settings),
-            SETTING_HTTP_DETAILED_ERRORS_ENABLED.get(settings),
-            SETTING_PIPELINING_MAX_EVENTS.get(settings),
```
Comment (pugnascotia) [inline server/src/main/java/org/elasticsearch/http/HttpHandlingSettings.java:75]: Why are these removed?

Comment (original-brownbear) [reply server/src/main/java/org/elasticsearch/http/HttpHandlingSettings.java:75]: These aren't used from this object, we load them into fields on the actual http transport and use them there.

**Review by pugnascotia — APPROVED** [review]
